sdk/python/kubeflow/training/utils/utils.py

```diff
@@ -230,13 +232,18 @@ def get_container_spec(
     # Add resources to the container spec.
     container_spec.resources = resources
 
+    # Add environment variables to the container spec.
+    container_spec.env = env
+    container_spec.env_from = env_from
+
     return container_spec
 
 
 def get_pod_template_spec(
     containers: List[models.V1Container],
     init_containers: Optional[List[models.V1Container]] = None,
     volumes: Optional[List[models.V1Volume]] = None,
+    restart_policy: Optional[str] = None,
```

Comment:
Do you require to set restart_policy when using PyTorchJob with Katib Experiment ?

Reply:
@andreyvelich We don't need to set `restart_policy` when using PyTorchJob with a Katib Experiment. However, since we are currently using batch job for the Katib experiment and reusing this function in Katib, the `restart_policy` must be either "Never" or "OnFailure." The default value of `restart_policy` in `V1PodSpec` is set to "Always," so I think we need to set it specifically here. What do you think?

Reply:
Do you want to use this helper function for Job submission as well ?
As you can see, right now we install `kubeflow-training` package with `kubeflow-katib` SDK only when user wants to use `tune` function with LLM capabilities: https://github.com/kubeflow/katib/pull/2393/files#diff-90679e5b4a47a4a806115400af64ef3dcaa5190a4e9b5c1f4cacd1626736bc38R73

Thus, when user just installs `kubeflow-katib` SDK and want to use `tune` API with Job, it will fail.

Can we discuss in the future how we can re-use helper functions from `kubeflow-training` SDK in `kubeflow-katib`?
For now, let's just focus on LLM API with `tune` function.

Reply:
I agree with your points. Currently, I only use this helper function when users work with external models and datasets. In those cases, users will need to install `extra_requires = {"huggingface": ["kubeflow-training[huggingface]"],}`. So, it should be fine.

However, I understand your concerns, and we can revisit this in the future. For now, I will close this PR and update the LLM API with the tune function to avoid reusing this helper function.